airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/base_streams.py

```diff
@@ -85,18 +85,29 @@ def failure(response: FacebookResponse, request: Optional[FacebookRequest] = Non
             requests_q.put(request)
 
         api_batch: FacebookAdsApiBatch = self._api.api.new_batch()
-
         while not requests_q.empty():
             request = requests_q.get()
             api_batch.add_request(request, success=success, failure=partial(failure, request=request))
-            if len(api_batch) == self.max_batch_size or requests_q.empty():
-                # make a call for every max_batch_size items or less if it is the last call
+            if requests_q.empty():
                 self._execute_batch(api_batch)
-                yield from records
-                records = []
                 api_batch: FacebookAdsApiBatch = self._api.api.new_batch()
+        return records
 
-        yield from records
+    def execute_in_batch(self, pending_requests: Iterable[FacebookRequest]) -> Iterable[MutableMapping[str, Any]]:
+        requests_q = Queue()
+        for r in pending_requests:
+            requests_q.put(r)
+        jobs = []
+        batch = []
+        while not requests_q.empty():
+            batch.append(requests_q.get())
+            # make a batch for every max_batch_size items or less if it is the last call
+            if len(batch) == self.max_batch_size or requests_q.empty():
```

Comment:
This is the original code from the repo, but it is very strange:
why not just take everything from pending_request (while splitting it in chunks of max_batch_size elements) and then run jobs.append? Why is there a need to go through a Queue?